File: src/realm/RealmServer.java
package realm;

import java.io.IOException;
import java.net.ServerSocket;
import java.util.Calendar;

import comunes.MainServidor;

public class RealmServer implements Runnable{

	private ServerSocket _SS;
	private Thread _t;
	
	public static int _totalNonAbo = 0;//Total de connections non abo
	public static int _totalAbo = 0;//Total de connections abo
	public static int _queueID = -1;//Numéro de la queue
	public static int _subscribe = 1;//File des non abonnées (0) ou abonnées (1)
	
	public RealmServer()
	{
		try {
			_SS = new ServerSocket(MainServidor.CONFIG_REALM_PORT);
			_t = new Thread(this);
			_t.setDaemon(true);
			_t.start();
		} catch (IOException e) {
			addToLog("IOException: "+e.getMessage());
			MainServidor.closeServers();
		}
		
	}

	public void run()
	{	
		while(MainServidor.isRunning)//bloque sur _SS.accept()
		{
			try
			{
				new RealmThread(_SS.accept());
			}catch(IOException e)
			{
				addToLog("IOException: "+e.getMessage());
				try
				{
					addToLog("Fermeture du serveur de connexion");	
					if(!_SS.isClosed())_SS.close();
				}
				catch(IOException e1){}
			}
		}
	}
	
	public void kickAll()
	{
		try {
			_SS.close();
		} catch (IOException e) {}
	}
	public synchronized static void addToLog(String str)
	{
		System.out.println(str);
		if(MainServidor.canLog)
		{
			try {
				String date = Calendar.HOUR_OF_DAY+":"+Calendar.MINUTE+":"+Calendar.SECOND;
				MainServidor.Log_Realm.write(date+": "+str);
				MainServidor.Log_Realm.newLine();
				MainServidor.Log_Realm.flush();
			} catch (IOException e) {}//ne devrait pas avoir lieu
		}
	}
	
	public synchronized static void addToSockLog(String str)
	{
		if(MainServidor.CONFIG_DEBUG)System.out.println(str);
		if(MainServidor.canLog)
		{
			try {
				String date = Calendar.HOUR_OF_DAY+":"+Calendar.MINUTE+":"+Calendar.SECOND;
				MainServidor.Log_RealmSock.write(date+": "+str);
				MainServidor.Log_RealmSock.newLine();
				MainServidor.Log_RealmSock.flush();
			} catch (IOException e) {}//ne devrait pas avoir lieu
		}
	}

	public Thread getThread()
	{
		return _t;
	}
}

File: src/realm/RealmServer.kt
package realm

import comunes.MainServidor
import comunes.MainServidor.closeServers
import java.io.IOException
import java.net.ServerSocket
import java.util.*

class RealmServer : Runnable {

    private var serversocket: ServerSocket? = null
    var thread: Thread? = null

    override fun run() {
        while (MainServidor.isRunning) {//bloque sur _SS.accept()
            try {
                RealmThread(serversocket!!.accept())
            } catch (e: IOException) {
                agregar_a_los_logs("IOException: " + e.message)
                try {
                    agregar_a_los_logs("Fermeture du serveur de connexion")
                    if (!serversocket!!.isClosed) serversocket!!.close()
                } catch (e1: IOException) {
                }
            }
        }
    }

    fun kickAll() {
        try {
            serversocket!!.close()
        } catch (e: IOException) {
        }
    }

    companion object {
        var totalnoabonado = 0 //Total de no abonados conectados
        @JvmField
		var totalabonado = 0 //Total de abonados conectados
        var idcola = -1 //Numero de id en cola
        var abonado = 1 //Fila de no abonados (0) - Fila de abonados (1)

        @JvmStatic
		@Synchronized
        fun agregar_a_los_logs(str: String) {
            println(str)
            if (MainServidor.canLog) {
                try {
                    val date = Calendar.HOUR_OF_DAY.toString() + ":" + Calendar.MINUTE + ":" + Calendar.SECOND
                    MainServidor.Log_Realm!!.write("$date: $str")
                    MainServidor.Log_Realm!!.newLine()
                    MainServidor.Log_Realm!!.flush()
                } catch (e: IOException) {
                } //ne devrait pas avoir lieu
            }
        }

        @JvmStatic
		@Synchronized
        fun addToSockLog(str: String) {
            if (MainServidor.MOSTRAR_ENVIADOS) println(str)
            if (MainServidor.canLog) {
                try {
                    val date = Calendar.HOUR_OF_DAY.toString() + ":" + Calendar.MINUTE + ":" + Calendar.SECOND
                    MainServidor.Log_RealmSock!!.write("$date: $str")
                    MainServidor.Log_RealmSock!!.newLine()
                    MainServidor.Log_RealmSock!!.flush()
                } catch (e: IOException) {
                } //ne devrait pas avoir lieu
            }
        }
    }

    init {
        try {
            serversocket = ServerSocket(MainServidor.PUERTO_DE_CONEXION)
            thread = Thread(this)
            thread!!.isDaemon = true
            thread!!.start()
        } catch (e: IOException) {
            agregar_a_los_logs("IOException: " + e.message)
            closeServers()
        }
    }
}